clustered/integration-test/src/test/java/org/ehcache/clustered/BasicCacheOpsMultiThreadedTest.java

```diff
@@ -0,0 +1,208 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.clustered;
+
+import org.ehcache.Cache;
+import org.ehcache.PersistentCacheManager;
+import org.ehcache.clustered.client.config.ClusteredStoreConfiguration;
+import org.ehcache.clustered.client.config.builders.ClusteredResourcePoolBuilder;
+import org.ehcache.clustered.client.config.builders.ClusteringServiceConfigurationBuilder;
+import org.ehcache.clustered.client.config.builders.ServerSideConfigurationBuilder;
+import org.ehcache.clustered.client.config.builders.TimeoutsBuilder;
+import org.ehcache.clustered.common.Consistency;
+import org.ehcache.config.ResourcePool;
+import org.ehcache.config.builders.CacheConfigurationBuilder;
+import org.ehcache.config.builders.CacheManagerBuilder;
+import org.ehcache.config.builders.ResourcePoolsBuilder;
+import org.ehcache.config.units.MemoryUnit;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.terracotta.testing.rules.Cluster;
+
+import com.tc.util.Assert;
+
+import java.io.File;
+import java.io.Serializable;
+import java.net.URI;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.terracotta.testing.rules.BasicExternalClusterBuilder.newCluster;
+
+public class BasicCacheOpsMultiThreadedTest extends ClusteredTests {
+
+  private static final String RESOURCE_CONFIG =
+    "<config xmlns:ohr='http://www.terracotta.org/config/offheap-resource'>"
+    + "<ohr:offheap-resources>"
+    + "<ohr:resource name=\"primary-server-resource\" unit=\"MB\">64</ohr:resource>"
+    + "</ohr:offheap-resources>" +
+    "</config>\n";
+
+  @ClassRule
+  public static Cluster CLUSTER =
+    newCluster().in(new File("build/cluster")).withServiceFragment(RESOURCE_CONFIG).build();
+
+  @BeforeClass
+  public static void waitForActive() throws Exception {
+    CLUSTER.getClusterControl().waitForActive();
+  }
+
+  private static final String CLUSTERED_CACHE_NAME    = "clustered-cache";
+  private static final String SYN_CACHE_NAME = "syn-cache";
+  private static final String PRIMARY_SERVER_RESOURCE_NAME = "primary-server-resource";
+  private static final String CACHE_MANAGER_NAME = "/crud-cm";
+  private static final int PRIMARY_SERVER_RESOURCE_SIZE = 4; //MB
+  private static final int NUM_THREADS = 20;
+
+  private AtomicReference<Throwable> exception = new AtomicReference<>();
+
+
+  @Test
+  public void testMulipleClients() throws Throwable {
```

Comment:
Can you explain what the test is doing in the javadoc? What's expected and why.

Reply:
added to the test class header